lib/ioc-volumes

```diff
@@ -0,0 +1,196 @@
+        _mount_options=$(echo "${_mount}" | cut -d: -f4)
+        _retval=0
+
+        if [ -d "${_source}" ] ; then
+            mount -t nullfs -o "${_mount_options},late" "${_source}" "${_full_dest}"
+            _retval=$?
+        else
+            cp -a "${_source}" "${_full_dest}"
+            if [ $? -eq 0 -a -n "$(echo ${_mount_options} | grep -E '(^|,)ro(,|$)')" ] ; then
+                chflags schg "${_full_dest}"
+                _retval=$?
+            fi
+        fi
+
+        if [ "${_retval}" -eq 0 ] ; then
+            _mounted="${_full_dest} ${_mounted}"
+            echo "    ${_source} -> ${_dest} (${_mount_options})"
+        else
+            echo "  ERROR: Mounting volume ${_source} at ${_full_dest} ${_mount_options} FAILED"
+
+            if [ -n "$_mounted" ] ; then
+                echo "  - Unmounting previously mounted volumes:"
+                for _mount in ${_mounted} ; do
+                    if [ -d "${_source}" ] ; then
+                        umount "${_mount}"
+                    else
+                        chflags noschg "${_mount}"
+                        rm "${_mount}"
+                    fi
+                    echo "    ${_mount}"
+                done
+            fi
+
+            __stop_jail ${_uuid} no_unmount
+            exit 1
+        fi
+    done
+}
+
+__unmount_volumes () {
+    local _uuid=$1
+    local _mountpoint _mount_list _source _dest _full_dest _retval
+    local volumes="$(__get_jail_prop volumes ${_uuid})"
+
+    if [ -z "${volumes}" -o "${volumes}" == "none" ] ; then
+        return
+    fi
+
+    _mountpoint="$(__get_jail_prop mountpoint ${_uuid})"
+    _mount_list="$(mount -v)"
+
+    echo "  + Unmounting volumes:"
+
+    # reverse volume order for unmounting correctly
+    volumes=$(echo $volumes | awk '{ for (i=NF; i>1; i--) printf("%s ",$i); print $1; }')
+
+    for _volume in ${volumes} ; do
+        _source=""
+        _dest=""
+        _retval=0
+
+        if [ -z "$(echo ${_volume} | grep ':')" ] ; then
+            _source=${_volume}
+        else
+            # split into source, dest and mount options
+            _source=$(echo ${_volume} | cut -d: -f1)
+            _dest=$(echo ${_volume} | cut -d: -f2)
+        fi
+
+        # set destination path to same as source if missing
+        if [ -z "${_dest}" ] ; then
+            _dest="${_source}"
+        fi
+
+        # add mointpoint
+        _full_dest="${_mountpoint}/root${_dest}"
+
+        if [ -d "${_source}" ] ; then
+            if [ -n "$(echo "${_mount_list}" | grep -F "${_full_dest}")" ] ; then
+                umount "${_full_dest}"
+                _retval=$?
+            else
+                echo "  WARNING: volume ${_source} was not mounted"
+                continue
+            fi
+        elif [ -f "${_full_dest}" ] ; then
+            chflags noschg "${_full_dest}"
+            rm "${_full_dest}"
+            _retval=$?
+        else
+            # not supported or nothing to do
```

Comment:
you did see the `continue` on the next line down, right? and you know that it instantly continues the for loop and therefor not executing the code further down? also i left this to maybe support other file types later if needed.